packages/medusa/src/loaders/module.ts

```diff
@@ -45,12 +40,11 @@ export default async ({
       trackInstallation(installation, "module")
     } catch (err) {
       if (resolution.definition.isRequired) {
-        throw new Error(
-          `Could not resolve required module: ${resolution.definition.label}`
-        )
+        logger?.warn(`Could not resolve module: ${resolution.definition.label}`)
```

Comment:
suggestion: I am unsure why the logger is optional, I would suggest to not make it optional